.github/workflows/build-runner.yml

```diff
@@ -0,0 +1,64 @@
+---
+"on":
+  pull_request:
+    branches:
+      - "**"
+    paths:
+      - "Dockerfile"
+      - .github/workflows/build-runner.yml
+  push:
+    branches:
+      - master
+    paths:
+      - "Dockerfile"
+      - .github/workflows/build-runner.yml
+
+name: Runner
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Build ${{ matrix.name }}
+    strategy:
+      matrix:
+        include:
+          - name: actions-runner
+            dockerfile: Dockerfile
```

Comment:
How do we use the matrix in this case? Do we have multiple builds? Maybe you're planning to have different builds in the future.

I see `matrix.name` is used below and that would resolve to `actions:runner` when tagging the image which is working, but you may consider have that hardcoded or in another environment variable called `IMAGE_NAME` or something like that, to be more explicit (but don't need to change anything, it's just an observation and potential readability improvement)